mesonbuild/linkers.py

```diff
@@ -1098,6 +1098,13 @@ def get_link_whole_for(self, args: T.List[str]) -> T.List[str]:
             return args
         return self._apply_prefix('--whole-archive') + args + self._apply_prefix('--no-whole-archive')
 
+    def get_pie_args(self) -> T.List[str]:
+        # Available in Solaris 11.2 and later
+        return ['-z', 'type=pie']
```

Comment:
It doesn't look like the version in openindiana has this option (although my vm is a little out of date). If they haven't added it, you'll need to check to see if this works before using it.

Reply:
You appear to be right about the illumos-based OS's.  I don't see any other examples of checking linker options in linkers.py - any suggestions on how to do this?

Reply:
Done in #7422